src/Hosting/TestHost/src/ClientHandler.cs

```diff
@@ -115,12 +128,15 @@ internal ClientHandler(PathString pathBase, ApplicationWrapper application)
                     }
                 }
 
-                if (body.CanSeek)
+                var requestPipe = new Pipe();
+
+                _ = Task.Run(async () =>
```

Comment:
This task should be awaited on the request processing side, here. https://github.com/aspnet/AspNetCore/blob/79ef2f6e472fe9e7d26562cca56cb38a44ae2e0e/src/Hosting/TestHost/src/HttpContextBuilder.cs#L99

It's also where you could complete the reader side of the pipe.